docs/getting-started/java.md

```diff
@@ -0,0 +1,254 @@
+
+        return openFeatureAPI;
+    }
+}
+```
+
+Next, we can add our REST endpoint definition with dependency injection for `OpenFeatureAPI` instance,
+
+```java
+package com.demo;
+
+import dev.openfeature.sdk.Client;
+import dev.openfeature.sdk.OpenFeatureAPI;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class RestHello {
+    private final OpenFeatureAPI openFeatureAPI;
+
+    @Autowired
+    public RestHello(OpenFeatureAPI OFApi) {
+        this.openFeatureAPI = OFApi;
+    }
+    
+    @GetMapping("/hello")
+    public String getHello() {
+        final Client client = openFeatureAPI.getClient();
+
+        // Evaluate welcome-message feature flag
+        if (client.getBooleanValue("welcome-message", false)) {
+            return "Hello, welcome to website.!";
+        }
+
+        return "Hello.!";
+    }
+}
+```
+
+At this point, we are ready to run the initial version of our application.
+
+### Step 4: Run the initial application
+
+Let's compile and run the application.
+
+<Tabs groupId="running">
+<TabItem value="maven" label="Maven">
+
+```shell
+mvn clean install
+java -jar target/demo-0.0.1-SNAPSHOT.jar
+```
+</TabItem>
+<TabItem value="gradle" label="Gradle">
+
+```shell
+./gradlew clean build
+java -jar build/libs/demo-0.0.1-SNAPSHOT.jar
+```
+</TabItem>
+</Tabs>
+
+> Note that you may have a different `.jar` name based on your build configurations.
+
+Now you can visit the url [http://localhost:8080/hello](http://localhost:8080/hello) and observe the message **Hello.!**.
+
+At this point, we are falling back to the default response. By default, OpenFeature will return the default value when
+there are no providers configured or running. In our code, we used flagd provider, yet we did not configure and run it. 
+
+### Step 5: Configure and run FlagD
+
+Create a new file named `flags.json` and add the following JSON. Notice that there's a flag called `welcome-message` which matches the flag
+key defined in [step 3](#step-3-add-code). The `welcome-message` flag has `on` and `off` variants that return `true` and `false` respectively.
+The state property controls whether the feature flag is active or not. Finally, the defaultVariant property controls the variant that should be returned. 
+In this case, the defaultVariant is `off`, therefore the value `false` would be returned.
+
+```json
+{
+  "$schema": "https://raw.githubusercontent.com/open-feature/schemas/main/json/flagd-definitions.json",
+  "flags": {
+    "welcome-message": {
+      "variants": {
+        "on": true,
+        "off": false
+      },
+      "state": "ENABLED",
+      "defaultVariant": "off"
+    }
+  }
+}
+```
+
+> NOTE: This configuration is specific for flagd and vary across other providers.
+
+Next, pull the latest flagd image,
+
+```shell
+docker pull ghcr.io/open-feature/flagd:latest
```

Comment:
I don't think this step in necessary. The Docker run command should fetch the latest image.

Reply:
Yes correct, will remove this part.